### PR TITLE
Use meson for C++ tests

### PR DESCRIPTION
**Pull request by arcondello**
Continuation of https://github.com/dwavesystems/dwave-optimization/pull/90

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -39,44 +41,55 @@ dwave_optimization_src = [
 if host_machine.system() == 'windows'
     # Shared libraries don't work so well on windows, so we build from a static
     # library instead
-    libdwave_optimization = static_library(
-        'dwave-optimization',
-        sources: dwave_optimization_src,
-        include_directories: dwave_optimization_include,
+    libdwave_optimization = declare_dependency(
```
Comment (arcondello) [inline meson.build:44]: `declare_dependency` just packages the library with the include for subsequent commands, so a bit tidier.

File: README.rst [hunk 2]
```diff
@@ -78,6 +78,35 @@ for more details.
...
+To run the Python tests, first install the package using an editable install
+as described above. The tests can then be run with
+`unittest <https://docs.python.org/3/library/unittest.html>`_.
+
+.. code-block:: bash
+
+    python -m unittest
+
+To run the C++ tests, first install the project dependencies, then setup a
+``meson`` build directory. You must configure the build as a debug build for
```
Comment (wbernoudy) [inline README.rst:95]: This refers to doing `meson setup _build -Dbuildtype=debug`, correct?

Comment (arcondello) [reply README.rst:95]: `meson setup build -Dbuildtype=debug` is what I use in the example here, but yes. Specifically the `-Dbuildtype=debug` bit.

Comment (wbernoudy) [reply README.rst:95]: Oops, missed the next couple of lines here :facepalm:

**Review by wbernoudy — APPROVED** [review]

**Pull request merged**